OneCommunityGlobal/HGNRest
# Jerry - Track Who Changes Permissions On Permissions Management Page

# Description
<img width="696" alt="description" src="https://github.com/OneCommunityGlobal/HGNRest/assets/55968519/de743455-2395-4448-a35e-2d2d6289218f">

Edit (Sep 21, 2023): Added a table on the Permissions Management page that fetches the permissionsChangeLogs from the dev database. Only Owner accounts can view see the table. Admin accounts are unable to see the table.


## Related PRS (if any):
* FE PR [#1277](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1277)

## Main changes explained:

- Added `permissionChangeLog` schema
- Created a logging function and a middleware function that adds a log to the `permissionChangeLogs` collection in the dev MongoDB database.
- Applied middleware function to PATCH `/roles/:roleId` route.

**Edit (Sep 21, 2023)**: 
* Created controller and router for operations on the `permissionChangeLog` collection.


## How to test:

1. check into current branch. 
2. do `npm install` and `...` to run this PR locally
3. Go to FE PR and follow “How to test” instructions there to start the frontend.


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/55968519/8fa9c22e-0fc3-46a4-a38c-955b9044201b


## Note:

It would be ideal for the `permissionChangeLogs` collection to be read-only and for the logs to have an expiration date (i.e., be automatically deleted after some set amount of time). From my understanding, these configurations can be made to the database by Jae or any trusted developer. If there it is possible/preferable to set these configurations in a Node.js environment, please let me know!
